### PR TITLE
Fluorite Octet

### PR DESCRIPTION
**Pull request by ghost**
fluorite

Adds the fluorite octet, magical 8 sided dice that have assorted effects. Adds it for 8 tc for mime traitors.
Rolls:
1 - Kill the user
2 - Spawn "creature" in all directions
3 - Destroy the user's equipment
4 - Random sized explosion
5 - Admin-heal the user
6 - Spawn a revolver
7 - Spawn a telecrystal
8 - Spawn a captain's spare ID

If you can think of cool effects, please comment, maybe we could push it to 16. Not only positive effects but negative ones too

:cl:
add: Adds the Fluorite Octet for Mime Traitors
/:cl:

### CODE REVIEW

**Comment by Repukan** [comment]
Should make it Gib if 1

Also it's too generous in what it can give.

**Comment by DaedalusGame** [comment]
>homestuck

**Comment by uraniummeltdown** [comment]
yes i know it's actually 8 d8s but thinking up 56 effects is a lot of work

**Comment by DaedalusGame** [comment]
you cannot run from your past
none of us can

File: code/game/objects/items/dice.dm [hunk 1]
```diff
@@ -157,6 +159,66 @@
...
+	if(!used)
+		if(!ishuman(user) || !user.mind || (user.mind in SSticker.mode.wizards))
+			to_chat(user, "<span class='warning'>You feel the magic of the dice is restricted to ordinary humans!</span>")
+			return
+		if(rigged)
+			eff3ct(user,rigged)
+		else
+			eff3ct(user,result)
+
+/obj/item/dice/fluorite/proc/eff3ct(mob/living/carbon/human/user, roll)
```
Comment (Jordie0608) [inline code/game/objects/items/dice.dm]: Degenerate proc naming.

File: code/game/objects/items/dice.dm [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
 /obj/item/storage/pill_bottle/dice/Initialize()
 	. = ..()
-	var/special_die = pick("1","2","fudge","space","00","8bd20","4dd6","100")
+	var/special_die = pick("1","2","fudge","space","00","8bd20","4dd6","100","fluorite")
```
Comment (Jordie0608) [inline code/game/objects/items/dice.dm]: You should really mention this part too.

File: code/game/objects/items/dice.dm [hunk 3]
```diff
@@ -157,6 +159,66 @@
...
+	..()
+	if(!used)
+		if(!ishuman(user) || !user.mind || (user.mind in SSticker.mode.wizards))
+			to_chat(user, "<span class='warning'>You feel the magic of the dice is restricted to ordinary humans!</span>")
+			return
+		if(rigged)
+			eff3ct(user,rigged)
+		else
+			eff3ct(user,result)
+
```
Comment (Jordie0608) [inline code/game/objects/items/dice.dm]: Needs a message when rolling a used dice, or some other indicator.

**Review by Jordie0608 — CHANGES_REQUESTED** [review]

**Comment by epochayur** [comment]
i mean it's a neat idea but jesus christ
>homestuck

**Comment by DaedalusGame** [comment]
all neatness is lost because the dice of fate is already this but implemented

File: code/modules/uplink/uplink_items.dm [hunk 4]
```diff
@@ -1217,6 +1217,14 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 	restricted_roles = list("Mime")
 	surplus = 0
 
+/datum/uplink_item/role_restricted/fluorite
+	name = "Fluorite Octet"
+	desc = "Rolling the dice will execute a wide range of highly unpredictable effects. Infinite uses."
+	cost = 8
```
Comment (nicbn) [inline code/modules/uplink/uplink_items.dm]: Too expensive for no direct advantage.

Comment (DaedalusGame) [reply code/modules/uplink/uplink_items.dm]: reminder that people will tc trade for the dice of fate and it's so good that admins won't give you the infinite version

**Comment by AlexTheSysop42** [comment]
> just porting the dice from SS413

At least add more effects for variety

**Review by Cyberboss — CHANGES_REQUESTED** [review]
Should have the correct 57 possible effects (8 - 64). At least half should be negative. Pad it with fluff if need be. Should do 8 rolls to properly get the result.

**Comment by DaedalusGame** [comment]
Objection. 64 - 8 = 56

**Comment by epochayur** [comment]
if you roll 2d6, that's possible numbers 2-16, but there's 15 results, not 14. Test it out for yourself

**Comment by Cyberboss** [comment]
It's not a subtraction, it's a range

**Comment by DaedalusGame** [comment]
ITT I've been owned and today we made apparant that bord never knows what he's talking about we apologize for the inconvenience

**Comment by Repukan** [comment]
Guess it's always past 7 PM for you huh

**Review by ChangelingRain — COMMENTED** [review]
[Do this properly, with all 8^8 possible combinations.](http://mspaintadventures.wikia.com/wiki/Dicekind#Powers)

**Comment by DaedalusGame** [comment]
>make a pr for a shitty homestuck meme
>maintainers understand the reference but because the pr is shitty they ask for the full thing to kill the pr

**Comment by ChangelingRain** [comment]
Well, why *wouldn't* you do it properly? Are you a *coward* or som8thing?

**Comment by Cyberboss** [comment]
8^8 is a meme. There's only 8^2 and she's a huge exaggerator

**Comment by Cyberboss** [comment]
Tempting though....

**Comment by KorPhaeron** [comment]
We already have a die of fate and we already have a vriska reference that is way more subtle so I am not sure what the point of this one is

**Comment by KorPhaeron** [comment]
Also why the mime?

**Comment by Cyberboss** [comment]
Probably as a rarity justification, which I don't think matters here.

Adding this to the uplink seems way out of character, this would be more suited to a necro chest.

At the same time
>buffing mining

**Comment by AlexTheSysop42** [comment]
>why the mime?

the justification when it was added to ss413 was that the mime didnt have as much uplink stuff (for reference, there was another homestuck reference clown traitor item added around that time)

>Well, why wouldn't you do it properly? Are you a coward or som8thing?

the 8^8 combinations thing would be literally impossible to code which is why it was changed to just 8, 64 would be a lot more interesting and actually be feasible

>we already have a vriska reference that is way more subtle

wait, where?

**Comment by DaedalusGame** [comment]
that's why its subtle

**Comment by KorPhaeron** [comment]
>wait, where?

The blue d8

**Comment by uraniummeltdown** [comment]
Going to pick this up l8r when I have time

**Pull request closed without merging**